herddb-core/src/main/java/herddb/cluster/EmbeddedBookie.java

```diff
@@ -188,16 +190,17 @@ private void dumpBookieConfiguration(Path bookie_dir, org.apache.bookkeeper.conf
 
     @Override
     public void close() {
-        if (bookieServer != null) {
+        if (embeddedServer != null) {
             LOG.info("Apache Bookkeeper stopping");
             try {
-                bookieServer.shutdown();
-
-                bookieServer.join();
+                embeddedServer.getLifecycleComponentStack().close();
+                if (waitForBookieServiceState(Lifecycle.State.CLOSED)) {
+                    LOG.info("Apache Bookkeeper stopped");
+                }
```

Comment:
what about adding a log line in case of timeout ?

Reply:
added a warning here and in the start() method